pyomo/contrib/doe/tests/test_doe_solve.py

```diff
@@ -0,0 +1,406 @@
+        FIM, Q, L, sigma_inv = get_FIM_Q_L(doe_obj=doe_obj)
+
+        # Since Trace is used, no comparison for FIM and L.T @ L
+
+        # Make sure FIM and Q.T @ sigma_inv @ Q are close (alternate definition of FIM)
+        self.assertTrue(np.all(np.isclose(FIM, Q.T @ sigma_inv @ Q)))
+
+    def test_reactor_fd_backward_solve(self):
+        fd_method = "backward"
+        obj_used = "trace"
+
+        experiment = FullReactorExperiment(data_ex, 10, 3)
+
+        DoE_args = get_standard_args(experiment, fd_method, obj_used)
+
+        doe_obj = DesignOfExperiments(**DoE_args)
+
+        doe_obj.run_doe()
+
+        self.assertTrue(doe_obj.results["Solver Status"] == "ok")
+
+        # self.assertTrue(that Q, F, and L are the same.
+        FIM, Q, L, sigma_inv = get_FIM_Q_L(doe_obj=doe_obj)
+
+        # Since Trace is used, no comparison for FIM and L.T @ L
+
+        # Make sure FIM and Q.T @ sigma_inv @ Q are close (alternate definition of FIM)
+        self.assertTrue(np.all(np.isclose(FIM, Q.T @ sigma_inv @ Q)))
+
+    def test_reactor_obj_det_solve(self):
+        fd_method = "central"
+        obj_used = "determinant"
+
+        experiment = FullReactorExperiment(data_ex, 10, 3)
+
+        DoE_args = get_standard_args(experiment, fd_method, obj_used)
+        DoE_args['scale_nominal_param_value'] = (
+            False  # Vanilla determinant solve needs this
+        )
+        DoE_args['_Cholesky_option'] = False
+        DoE_args['_only_compute_fim_lower'] = False
+
+        doe_obj = DesignOfExperiments(**DoE_args)
+
+        doe_obj.run_doe()
+
+        self.assertTrue(doe_obj.results['Solver Status'] == "ok")
+
+    def test_reactor_obj_cholesky_solve(self):
+        fd_method = "central"
+        obj_used = "determinant"
+
+        experiment = FullReactorExperiment(data_ex, 10, 3)
+
+        DoE_args = get_standard_args(experiment, fd_method, obj_used)
+
+        doe_obj = DesignOfExperiments(**DoE_args)
+
+        doe_obj.run_doe()
+
+        self.assertTrue(doe_obj.results["Solver Status"] == "ok")
+
+        # self.assertTrue(that Q, F, and L are the same.
+        FIM, Q, L, sigma_inv = get_FIM_Q_L(doe_obj=doe_obj)
+
+        # Since Cholesky is used, there is comparison for FIM and L.T @ L
+        self.assertTrue(np.all(np.isclose(FIM, L @ L.T)))
+
+        # Make sure FIM and Q.T @ sigma_inv @ Q are close (alternate definition of FIM)
+        self.assertTrue(np.all(np.isclose(FIM, Q.T @ sigma_inv @ Q)))
+
+    def test_compute_FIM_seq_centr(self):
+        fd_method = "central"
+        obj_used = "determinant"
+
+        experiment = FullReactorExperiment(data_ex, 10, 3)
+
+        DoE_args = get_standard_args(experiment, fd_method, obj_used)
+
+        doe_obj = DesignOfExperiments(**DoE_args)
+
+        doe_obj.compute_FIM(method="sequential")
+
+    def test_compute_FIM_seq_forward(self):
+        fd_method = "forward"
+        obj_used = "determinant"
+
+        experiment = FullReactorExperiment(data_ex, 10, 3)
```

Comment:
There are no assertions on either of these - is that intentional?

Reply:
This is intentional. The idea is to make sure the problem solves. I can add a check to make sure the output is what we expect, but that will take me a bit of time.

The four tests that solve a problem and have no assertions will become more "test-like" when we add the safeguarding for model solving noted with the immediate TODO's in #3345.

Reply:
That's fine - not all tests need asserts. It might be worth putting a comment summarizing this comment, however:
"These tests ensure that the options run without errors."